Civi/Core/SettingsManager.php

```diff
@@ -239,13 +280,52 @@ protected function getDefaults($entity) {
    */
   protected function getMandatory($entity) {
     if ($this->mandatory === NULL) {
-      $this->mandatory = self::parseMandatorySettings($GLOBALS['civicrm_setting'] ?? NULL);
+      $this->mandatory = self::parseMandatorySettingsGlobalVar($GLOBALS['civicrm_setting'] ?? NULL);
+
+      // merge in settings from env - these take precedence over values from global
+      // TODO: should we warn if env var is overriding $civicrm_setting setting?
+      foreach (['domain', 'contact'] as $entityKey) {
+        $this->mandatory[$entityKey] = array_merge(
+            $this->mandatory[$entityKey] ?? [],
+            $this->getEnvSettingValues($entity)
+        );
+      }
     }
     return $this->mandatory[$entity];
   }
 
   /**
-   * Parse mandatory settings.
+   * Get any setting values set using environment variables
+   *
+   * @param string $entity
+   *   Ex: 'domain' or 'contact'.
+   *
+   * @return array
+   *   Array(string $settingName or $settingFqn => mixed $value).
+   */
+  protected function getEnvSettingValues($entity) {
+    $settings = [];
+
+    $specs = SettingsMetadata::getMetadata([
+      'is_contact' => ($entity === 'contact' ? 1 : 0),
+      'load_from_env' => TRUE,
+    ], NULL, FALSE, $this->preBoot);
+
+    foreach ($specs as $key => $spec) {
+      $fqn = $spec['global_name'] ?? NULL;
+      if ($fqn) {
+        $envValue = getenv($fqn);
+        if ($envValue) {
+          $settings[$key] = $envValue;
+        }
```

Comment:
Yes.

At this point I think that shouldn't cause trouble, because all the `load_from_env` settings should be truthy strings. But in building out it would be good to add some more intelligent casting / handling. But there's decisions to be made there, which I thought would be easiest to leave out for now. (In the previous PR I had (set FALSE for "0" "false" "FALSE" //  ignore "" // set the string value otherwise. It might be useful that we can consult the "type" from the settings meta at this point.)

Reply:
* +1 for allowing literal `"0"` or `0` to be respected as concrete values (*booleans*).
    * Since `getenv` [returns `FALSE` for undefined values](https://www.php.net/getenv), this just needs to check  `$envValues !== FALSE` 
    * This makes it *possible* to specify booleans as true or false (`SOME_VAR=1` or `SOME_VAR=0`).
    * We don't need to hold the whole PR on this subissue. (It can be a small/quick follow-up.)
* +1 for leaving out handling for `"true"` / `"false"` -- at least for now. There's sufficient expressiveness with `"0"`/`"1"`, and this allows a *simple* value-mapping.
    * Of course, it's not perfect. Some percentage of sysadmins would feel more comfortable with `SOME_VAR=true`/`SOME_VAR=false` than with `SOME_VAR=1`/`SOME_VAR=0`.
    * If we really want to sugar-coat the env-vars, then I think that moves us to type-dependent parsing... `MY_BOOL=false` and `MY_STR=false` need to parse differently.
    * I don't have a full opinion on whether it's worth the complication, but...
    * IMHO it's safe to add that later. We're not painted into a corner. (*Any env-vars that are legit under today's regime should be equally legit under a future/type-dependent regime.*)